.github/workflows/nix.yaml

```diff
@@ -0,0 +1,38 @@
+name: "Nix flake checks"
+on:
+  pull_request:
+  push:
+jobs:
+  skip_duplicate_jobs:
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          concurrent_skipping: 'same_content'
+          skip_after_successful_duplicate: 'true'
+          do_not_skip: '["pull_request", "workflow_dispatch", "schedule"]'
+  linux-checks:
+    needs: skip_duplicate_jobs
+    if: ${{ needs.skip_duplicate_jobs.outputs.should_skip != 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2.4.0
+    - uses: cachix/install-nix-action@v15
+      with:
+        extra_nix_config: |
+          access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+    - run: nix flake check
+  darwin-checks:
+    needs: skip_duplicate_jobs
+    if: ${{ needs.skip_duplicate_jobs.outputs.should_skip != 'true' }}
+    runs-on: macOS-latest 
+    steps:
+    - uses: actions/checkout@v2.4.0
+    - uses: cachix/install-nix-action@v15
+      with:
+        extra_nix_config: |
+          access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+    - run: nix flake check
```

Comment:
Missed this at first, but if you make this based on a matrix instead (example: [declared](https://github.com/rivosinc/hubris/blob/422408c02f03efea24ecb7556b6e8ec8bb148576/.github/workflows/dist.yml#L22-L24), [used](https://github.com/rivosinc/hubris/blob/422408c02f03efea24ecb7556b6e8ec8bb148576/.github/workflows/dist.yml#L141)) the Linux and MacOS runs can be parallelized, and you don't need to duplicate the job.

Reply:
Done.